### ISSUE TITLE
Parser should throw syntax error on assignment to a call expression

### ISSUE THREAD

**Issue by golopot** [typescript-eslint/typescript-eslint#1754]
**What code were you trying to parse?**
```ts
f(1) = 2
```

**What did you expect to happen?**
Syntax error.

**What actually happened?**
No error.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.24.0` |

**Comment by bradzacher** [comment]
This is syntactically valid code, thus it parses fine.
Typescript has its own error for this problem:
`The left-hand side of an assignment expression must be a variable or a property access.(2364)`

We do not duplicate typescript semantic errors, and only throw if there are syntax errors.

**Comment by golopot** [comment]
Both `acorn`, `espree` and `@babel/core` reject this as a syntax error, so I thought this parser should reject it too, to align with the behavior. But indeed the code seems to be syntactically valid, arguably it is more correct to not throw.
